addon/-private/model-factory.ts

```diff
@@ -2,33 +2,40 @@ import { getOwner } from '@ember/application';
 import { Dict } from '@orbit/utils';
 import { RecordIdentity, cloneRecordIdentity } from '@orbit/data';
 import Store from './store';
-import Model from './model';
+import Model, { ModelSettings } from './model';
+
+interface Factory {
+  create(settings: ModelSettings): Model;
+}
 
 export default class ModelFactory {
-  private _store: Store;
-  private _modelFactoryMap: Dict<any>;
+  #store: Store;
+  #modelFactoryMap: Dict<Factory>;
+  #mutableModels: boolean;
```

Comment:
How about skipping `#mutableModels` here in the `ModelFactory` and instead allowing models to access a `mutableModels` property from their `#store`? If each `Store` had a `mutableModels` getter / setter it could control mutability for all its associated models with a single boolean.

Users could then enable mutable models like this:

```
store.mutableModels = true;
```

For consistency, it would also be nice to be able to pass this option when forking a store:

```
let readOnlyFork = store.fork({ mutableModels: false });
```

Does this seem like too much of a footgun to you?

Reply:
My original thought was to make this option as hard to find as possible. I really don't think it should be used except for migration from older versions and maybe quick demos. But even for quick demos I think we can expose some APIs around fork to make it better. This is why I did not want to expose it on the store or make it flipable on runtime. But if you feel different let me know and I will change it.